packages/xo-server-netbox/src/index.js

```diff
@@ -414,24 +414,24 @@ class Netbox {
     const vmsToDelete = nbVmsList.filter(nbVm => nbVm.custom_fields.uuid == null).map(nbVm => ({ id: nbVm.id }))
     const vmsToUpdate = []
     const vmsToCreate = []
-    for (const poolId of pools) {
+    for (const xoPoolId of xoPools) {
       // Get XO VMs that are on this pool
-      const poolVms = this.getObjects({ filter: { type: 'VM', $pool: poolId } })
+      const xoPoolVms = this.getObjects({ filter: { type: 'VM', $pool: xoPoolId } })
 
-      const nbCluster = nbClusters[poolId]
+      const nbCluster = nbClusters[xoPoolId]
 
       // Get Netbox VMs that are supposed to be in this pool
-      const poolNbVms = pickBy(nbVms, nbVm => nbVm.cluster.id === nbCluster.id)
+      const xoPoolNbVms = pickBy(nbVms, nbVm => nbVm.cluster.id === nbCluster.id)
```

Comment:
A bit weird :sweat_smile: 

Maybe `nbClusterVms`? :thinking:

Reply:
No, because that would mean the Netbox VMs that are in the Netbox cluster, but it's actually the Netbox VMs that are bound to the XO Pool. It might not be the same if there have been some changes since the last synchronization.